internal/pkg/csumio/csumreader_test.go

```diff
@@ -184,3 +184,191 @@ func TestReader_CorrectChecksum_VerificationEnabled(t *testing.T) {
 		})
 	}
 }
+
+// TestReader_ReadFactorCoverage exhaustively validates csumio.Reader.Read.
+// The test matrix uses pairwise coverage across three factors:
+//
+//	Factor A: requested len(p) relative to chunkSize (> chunk / == chunk / < chunk).
+//	Factor B: remaining unread bytes relative to len(p) (> p / == p / < p).
+//	Factor C: offset state (aligned at chunk boundary vs. mid-chunk).
+//
+// These three factors yield nine cases (case01–case09). Each entry below sets the total payload size,
+// pre-consumes bytes when needed, and issues a single Read with len(p) = chunkSize±1 so that the
+// target call experiences the intended combination while checksum verification remains enabled.
```

Comment:
How about writing down the nine cases in a table like this?

|len(p)|remaining bytes in the dataReader|reader's buffer|
|:----|:---|:---|
|> chunk size|> p|empty|
|> chunk size|== p|not empty|
|> chunk size|< p|empty|
|== chunk size|> p|empty|
|== chunk size|== p|not empty|
|== chunk size|< p|empty|
|< chunk size|> p|empty|
|< chunk size|== p|not empty|
|< chunk size|< p|empty|

 Then we can understand nine tests are enough or not. In addition, we can unify redundant `name` and `description` fields (see later).